regression/utils.py

```diff
@@ -3,12 +3,191 @@
 
 """Utility and helper functions for regression testing."""
 
+import logging
 import os
 import tempfile
+import urllib.parse
+from typing import Any, Dict
+
+logger = logging.getLogger(__name__)
+
+
+def get_kwargs(kwargs: str) -> str:
+    """Parse key of named command-line arguments.
+
+    Args:
+        kwargs (str): Command-line arguments.
+
+    Returns:
+        str: Key of named arguments.
+    """
+    if kwargs.startswith('--'):
+        kwargs = kwargs[2:]
+    kwargs = kwargs.replace('-', '_')
+    return kwargs
+
+
+def get_streaming_dataset_params(kwargs: Dict[str, str]) -> Dict[str, Any]:
```

Comment:
Is there a more scalable way to do argument parsing that also takes into account future changes in dataset/dataloader/writer?

Reply:
I have looked into multiple options, this is the best way I could find without overriding the defaults in dataset/dataloader/writer!!